metricsaggregator/src/main/java/io/dockstore/metricsaggregator/helper/ValidationStatusAggregator.java

```diff
@@ -0,0 +1,209 @@
+                        ValidatorInfo validationInfoForValidatorTool = entry.getValue();
+                        return validationInfoForValidatorTool.getValidatorVersions().stream();
+                    }, Collectors.toList())));
+
+            // For each validator tool, find the most recent ValidatorVersionInfo for each version
+            validatorToolToValidationVersionInfos.forEach((validatorTool, validationVersionInfosByValidatorTool) -> {
+                // Number of runs across all versions
+                final int numberOfRuns = validationVersionInfosByValidatorTool.stream().map(ValidatorVersionInfo::getNumberOfRuns)
+                        .mapToInt(Integer::intValue)
+                        .sum();
+                final List<DoubleStatistics> validationRunsStatistics = validationVersionInfosByValidatorTool.stream()
+                        .map(validatorVersionInfo -> new DoubleStatistics(validatorVersionInfo.getPassingRate(), validatorVersionInfo.getNumberOfRuns()))
+                        .toList();
+
+                final double passingRate = DoubleStatistics.createFromStatistics(validationRunsStatistics).getAverage();
+                final Optional<ValidatorVersionInfo> mostRecentValidationVersion = getLatestValidationVersionInfo(validationVersionInfosByValidatorTool);
+
+                if (mostRecentValidationVersion.isPresent()) {
+                    // Group ValidatorVersionInfo by version name
+                    Map<String, List<ValidatorVersionInfo>> versionNameToValidationVersionInfos = validationVersionInfosByValidatorTool.stream()
+                            .collect(Collectors.groupingBy(ValidatorVersionInfo::getName));
+
+                    // Get a list of the most recent ValidatorVersionInfo for each version
+                    List<ValidatorVersionInfo> mostRecentValidationVersionInfos = versionNameToValidationVersionInfos.values().stream().map(this::getLatestValidationVersionInfo).filter(Optional::isPresent).map(Optional::get).toList();
+
+                    // Set validation info for the validator tool
+                    ValidatorInfo validatorInfo = new ValidatorInfo()
+                            .mostRecentVersionName(mostRecentValidationVersion.get().getName())
+                            .validatorVersions(mostRecentValidationVersionInfos)
+                            .numberOfRuns(numberOfRuns)
+                            .passingRate(passingRate);
+
+                    newValidatorToolToValidatorInfo.put(validatorTool, validatorInfo);
+                }
+            });
+        }
+
+        if (newValidatorToolToValidatorInfo.isEmpty()) {
+            return Optional.empty();
+        }
+        return Optional.of(new ValidationStatusMetric().validatorTools(newValidatorToolToValidatorInfo));
+    }
+
+    static Optional<ValidationExecution> getLatestValidationExecution(List<ValidationExecution> executions) {
```

Comment:
these two methods look the same, can `ValidationExecution` and `ValidatorVersionInfo` share an interface that has `getDateExecuted`